src/main/java/edu/project3/logWorkers/LogAnalyser.kt

```diff
@@ -0,0 +1,156 @@
+package edu.project3.logWorkers
+
+import edu.project3.Table
+import jakarta.validation.constraints.NotNull
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+import kotlin.math.min
+
+class LogAnalyser {
+
+    fun getRequestsAmount(@NotNull logs: Table): Int =
+        logs.getSize()
+
+
+    fun getTheMostPopularResources(
+        logs: Table,
+        amount: Int = Int.MAX_VALUE,
+        request: String = "GET"
+    ): Table {
+        val sortedLogs = logs.getRows()
+            .filter { it["request_type"] != null }
+            .filter { it["request_type"] == request }
+            .map { it["request"] }
+            .groupingBy { it }
+            .eachCount()
+            .toList()
+            .sortedByDescending { it.second }
+
+        return Table(sortedLogs
+            .subList(0, min(sortedLogs.size, amount))
+            .map { mapOf("resource" to it.first!!, "value" to it.second.toString()) })
```

Comment:
переноси на разные строчки, 
```suggestion
            .map {
             mapOf("resource" to it.first!!, 
            "value" to it.second.toString()) 
            })
``` А то прям тяжело читать